ydb/core/tx/schemeshard/schemeshard__table_stats.cpp

```diff
@@ -5,6 +5,23 @@
 #include <ydb/core/base/cputime.h>
 #include <ydb/core/protos/sys_view.pb.h>
 
+namespace {
+
+TVector<TString> MapChannelsToStoragePoolKinds(const NKikimr::TStoragePools& pools, const NKikimr::TChannelsBindings& bindings) {
+    THashMap<TString, TString> nameToKindMap(pools.size());
+    for (const auto& pool : pools) {
+        nameToKindMap.emplace(pool.GetName(), pool.GetKind());
+    }
+    TVector<TString> channelsMapping;
+    channelsMapping.reserve(bindings.size());
+    for (const auto& binding : bindings) {
+        channelsMapping.emplace_back(nameToKindMap.at(binding.GetStoragePoolName()));
```

Comment:
@snaury Could it happen that we have received a channels' binding of some datashard that contains some storage pool name that is not present in the database storage pools array? Is it ok to leave .at (and potentially throw an error) here?

Reply:
We don't support removing storage pools right now, and schemeshard should only add storage pools that it knows about. However I'm not sure what could happen in some legacy databases, where `/Root` doesn't have any storage pools, but where it has tables. I suspect such tablets wouldn't have channel bindings in such cases, but I'm not sure, they could maybe have storage pool copied from `channels.txt`. I would prefer you don't crash when possible, and simply don't count such unknown channels towards per-pool-kind quota.